app/_features/event/components/event-date-picker.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  useDisclosure,
+} from '@nextui-org/react';
+import DatePicker from 'react-datepicker';
+import { useCallback, useEffect, useState } from 'react';
+
+interface DatePickerModalProps {
+  startDate: Date;
+  setStartDate: React.Dispatch<React.SetStateAction<Date>>;
+}
+
+export function DatePickerModal({
+  startDate,
+  setStartDate,
+}: DatePickerModalProps) {
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const { isOpen, onOpen, onClose, onOpenChange } = useDisclosure();
+
+  const openModal = useCallback(() => {
+    onOpen();
+  }, [onOpen]);
+
+  const onConfirm = useCallback(() => {
+    setStartDate(selectedDate);
+    onClose();
+  }, [onClose, selectedDate, setStartDate]);
+
+  const onCancel = useCallback(() => {
+    setSelectedDate(startDate);
+    onClose();
+  }, [onClose, startDate]);
+
+  useEffect(() => {
+    document.addEventListener('open:event-date-picker-modal', openModal);
+  });
+
+  const currentDate = new Date();
+  const yesterday = new Date(currentDate.setDate(currentDate.getDate() - 1));
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpenChange={onOpenChange}
+      placement="auto"
+      hideCloseButton
+      isDismissable={false}
+    >
+      <ModalContent>
+        <ModalHeader>Set event date</ModalHeader>
+        <ModalBody>
+          <DatePicker
+            selected={selectedDate}
+            onChange={(date) => {
+              if (date) {
+                setSelectedDate(date);
+                console.log(setSelectedDate);
```

Comment:
No need to log the selected date

Reply:
i've removed the logging https://github.com/inlivedev/inlive-room/pull/130/commits/06de4972dc0b8b36258652c019bd4a9acd9245ff#diff-e67452b9ca88fae0ce442b172f51c8f3721a58438e933fd32db8bcd31006655aL62